src/app/projects/components/project-settings/components/embeddings/embeddings.component.ts

```diff
@@ -104,25 +154,50 @@ export class EmbeddingsComponent implements OnInit, OnDestroy, OnChanges {
   addEmbedding() {
     if (!this.dataHandlerHelper.canCreateEmbedding(this.settingModals, this.embeddings, this.attributes)) return;
     const embeddingForm = this.settingModals.embedding.create.embeddingCreationFormGroup;
-    const embeddingHandle = embeddingForm.get("embeddingHandle").value;
     const attributeId = embeddingForm.get("targetAttribute").value;
-    const granularity = embeddingForm.get("granularity").value;
+    const platform = embeddingForm.get("platform").value;
+
+    if (this.gdprText) {
+      this.gdprText.nativeElement.innerHTML = this.gdprText.nativeElement.innerHTML.split(':')[0] + ":" + this.anchorGdpr.nativeElement.innerHTML;
```

Comment:
changing an innerHTML isn't recommended for text changes. Why not create a variable gdprTextHTML or something like that and fill it here + in HTML file add an *if for if the variable is filled?

Reply:
Yes we can add the variable here so we would avoid changing the innerHTML.
I think the check in the template is not needed since there we have a check if we want to display whole 'terms and conditions' part and here we check if that is displayed we take the message from there.

Maybe we could add the messages as constants and display them in the template and use them here instead of the whole implementation with the innerHTML text.

Reply:
i think the text were specifically implemented to be transferred from backend so having a constant in the UI wouldn't make sense. However I'm not sure if that's what you meant

Reply:
Okay yes, that's what I meant. Since it is transferred from the BE, the innerHTML and the constants will be removed.